### PR TITLE
Support for handling HTTPS using ESW certificates

### PR DESCRIPTION
**Pull request by arturek**

### CODE REVIEW

File: src/Eshopworld.Web.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.26730.16
+# Visual Studio Version 16
```
Comment (VaclavK) [inline src/Eshopworld.Web.sln]: hmmmm .. I do not think so... 16=2019

File: src/Eshopworld.Web/Eshopworld.Web.csproj [hunk 2]
```diff
@@ -18,15 +18,21 @@
     <IncludeSource>True</IncludeSource>
     <GeneratePackageOnBuild>false</GeneratePackageOnBuild>
     <DebugType>Full</DebugType>
+    <RuntimeIdentifier>win7-x64</RuntimeIdentifier>
+    <Platform>x64</Platform>
+    <Platforms>x64</Platforms>
```
Comment (VaclavK) [inline src/Eshopworld.Web/Eshopworld.Web.csproj]: this should not be necessary me thinks

File: src/Eshopworld.Web/Eshopworld.Web.csproj [hunk 3]
```diff
@@ -1,15 +1,15 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <Version>1.1.1</Version>
-    <PackageReleaseNotes>Aligned with the new domain event structure.</PackageReleaseNotes>
+    <Version>2.0.0</Version>
```
Comment (VaclavK) [inline src/Eshopworld.Web/Eshopworld.Web.csproj]: why major?

File: src/Eshopworld.Web/Eshopworld.Web.csproj (hunk 2 above)
Comment (dguerin) [reply src/Eshopworld.Web/Eshopworld.Web.csproj]: I think they are at the moment based on who consumes these values sadly. @arturek should have more details on this.

Comment (djfr) [reply src/Eshopworld.Web/Eshopworld.Web.csproj]: Packages should only target MSIL and they don't need any RID's. What was the rationale behind adding this?

File: src/Eshopworld.Web.sln (hunk 1 above)
Comment (djfr) [reply src/Eshopworld.Web.sln]: It doesn't matter, that is a hint for the global SLN opener to pick the right version. Since we all should have 2019 by now, why not?

File: src/Eshopworld.Web/Eshopworld.Web.csproj (hunk 3 above)
Comment (djfr) [reply src/Eshopworld.Web/Eshopworld.Web.csproj]: definitely not major, if we keep this pace on we'll be on version 10 by Q4 2019 😄

File: src/Eshopworld.Web.sln [hunk 4]
```diff
@@ -17,18 +17,18 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Eshopworld.Web.Tests", "Tes
...
-		{7B310A41-8B68-40C2-A0AF-AFBEF1677842}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{7B310A41-8B68-40C2-A0AF-AFBEF1677842}.Release|Any CPU.Build.0 = Release|Any CPU
+		{99B585E3-CDD0-42D3-8B70-90AA906F3CA6}.Debug|x64.ActiveCfg = Debug|x64
+		{99B585E3-CDD0-42D3-8B70-90AA906F3CA6}.Debug|x64.Build.0 = Debug|x64
+		{99B585E3-CDD0-42D3-8B70-90AA906F3CA6}.Release|x64.ActiveCfg = Release|x64
+		{99B585E3-CDD0-42D3-8B70-90AA906F3CA6}.Release|x64.Build.0 = Release|x64
+		{7B310A41-8B68-40C2-A0AF-AFBEF1677842}.Debug|x64.ActiveCfg = Debug|x64
+		{7B310A41-8B68-40C2-A0AF-AFBEF1677842}.Debug|x64.Build.0 = Debug|x64
+		{7B310A41-8B68-40C2-A0AF-AFBEF1677842}.Release|x64.ActiveCfg = Release|x64
+		{7B310A41-8B68-40C2-A0AF-AFBEF1677842}.Release|x64.Build.0 = Release|x64
```
Comment (djfr) [inline src/Eshopworld.Web.sln]: Packages are debuged in MSIL (Any CPU), so why did you remove that from the SLN configuration and just added x64?

File: src/Eshopworld.Web.sln (hunk 1 above)
Comment (VaclavK) [reply src/Eshopworld.Web.sln]: I hear you but in principle other teams can open this etc... not a biggie....more pedantic than anything

File: src/Eshopworld.Web/Eshopworld.Web.csproj (hunk 3 above)
Comment (dguerin) [reply src/Eshopworld.Web/Eshopworld.Web.csproj]: yeah I agree for now but I would like to follow a semver structure at some point such that other developers can understand the suggested impact the package will have before they consume it. Like @VaclavK mentioned before this could be something like when Sierra V1 is released or some other arbitrary event.

Comment (VaclavK) [reply src/Eshopworld.Web/Eshopworld.Web.csproj]: what I suggested is to go pre-release and then go for full release at appropriate moment.. I think it is cleanest

Comment (djfr) [reply src/Eshopworld.Web/Eshopworld.Web.csproj]: The impact is none since we just added a UseEshopworldSsl method to it. So it's an extra feature not changes in anything that is already present.

The way to properly do this is through Roslyn Analyzers, where we push a new feature and we start actively suggesting developers to use it through Roslyn. However that requires _precious_ time :)

File: src/Eshopworld.Web.sln (hunk 1 above)
Comment (arturek) [reply src/Eshopworld.Web.sln]: They can open it with any version later than MinimumVisualStudioVersion. This version is used only when the solution is opened for the first time and even then it's just a hint.

File: src/Eshopworld.Web/Eshopworld.Web.csproj (hunk 2 above)
Comment (arturek) [reply src/Eshopworld.Web/Eshopworld.Web.csproj]: ServiceFabric. And I don't know why their packages are compiled with this requirement.

File: src/Eshopworld.Web/Eshopworld.Web.csproj (hunk 3 above)
Comment (arturek) [reply src/Eshopworld.Web/Eshopworld.Web.csproj]: I'm using Semantic versioning rules, which states to increment "MAJOR version when you make incompatible API changes". The package has new dependencies so it's not compatible with its previous version. Last time when we'd sneakily introduced braking change without incrementing the major version I've lost some time trying to find out why my code doesn't work correctly any more. And I was not the only one affected. In other words I care more about developer experience and unsurprising process of package upgrade than about keeping version numbers as low as possible. Using pre-release version is possible although I think it doesn't help in our case too much.

File: src/Eshopworld.Web.sln (hunk 4 above)
Comment (arturek) [reply src/Eshopworld.Web.sln]: The Eshopworld.Web.csproj project requires it, so I haven't seen a reason why packages which depend on it should use other configuration.

File: src/Eshopworld.Web/Eshopworld.Web.csproj (hunk 3 above)
Comment (djfr) [reply src/Eshopworld.Web/Eshopworld.Web.csproj]: _Incompatibility_ means breaking. Adding a new package dependency won't break anything, at worst it introduces one more dependency that requires proper management of versions.

We had a problem in telemetry, because of two separate reasons - that go down to bad usage, not the introduction of breaking change on intended usage:
- Folks don't update everything, they think they can cherry pick whatever they update
- Creative folks decided to write their own BigBrother, and they didn't even inherit, so there isn't much we can do there except Analyzers.

File: src/Eshopworld.Web/Eshopworld.Web.csproj (hunk 2 above)
Comment (arturek) [reply src/Eshopworld.Web/Eshopworld.Web.csproj]: @VaclavK singular and plural are used for different proposes unfortunately. One by msbuild, the other mostly by VS.

File: src/Eshopworld.Web/Eshopworld.Web.csproj (hunk 3 above)
Comment (djfr) [reply src/Eshopworld.Web/Eshopworld.Web.csproj]: Agreed with major version, because this is a milestone for the package as we are now bound to ServiceFabric on the Web package.

Comment (VaclavK) [reply src/Eshopworld.Web/Eshopworld.Web.csproj]: and the strategy with subsequent updates? vote?
